charmcraft/charm_builder.py

```diff
@@ -240,6 +240,28 @@ def _install_dependencies(self, staging_venv_dir):
         # create virtualenv using the host environment python
         with instrum.Timer("Creating venv"):
             _process_run(["python3", "-m", "venv", str(staging_venv_dir)])
+
+        python_cmd = str(_find_venv_bin(staging_venv_dir, "python3"))
```

Comment:
can you explain why this new bit is necessary?

Reply:
This is another bug:
When it creating the `venv`, `pip` think the `pip` is already installed, and point it to the snapcraft pip. So the pip commands missing in the venv. And It has to be force reinstall to make it work.

Reply:
you mean that after that `["python3", "-m", "venv", str(staging_venv_dir)]` command, `pip` does *not* exist in `staging_venv_dir`?

Reply:
Yes

Reply:
This is super weird, that's pretty much what [craft-parts does](https://github.com/canonical/craft-parts/blob/main/craft_parts/plugins/python_plugin.py#L95) and it worked fine for snapcraft and rockcraft. How can pip not be there??

Reply:
The process are, create the basic python link, then use `ensurepip` to call pip to install another pip from wheel. However, due to how `snapcraft`(?) set the env, the pip found the pip in the snapcraft's venv, and skipped the install.

Reply:
weird. As an experiment, you can try "unstaging" `bin/pip` from the charmcraft snap, and then there wouldn't be a `pip`in the snapcraft venv